src/screens/Settings/AppIconSettings.tsx

```diff
@@ -1,252 +1,182 @@
-                  })}
-                  style={[
-                    {width: 100, height: 100},
-                    platform({
-                      ios: {borderRadius: 20},
-                      android: a.rounded_full,
-                    }),
-                    a.curve_continuous,
-                    a.shadow_lg,
-                  ]}
-                  accessibilityIgnoresInvertColors
-                />
-              </PressableScale>
-              <Text
-                style={[a.text_center, a.font_bold, a.text_md, a.mt_md]}
-                // for Classic™
-                emoji>
-                {icon.name}
-              </Text>
-            </View>
+        </Group>
+        <Text style={[a.text_xl, a.mt_lg, a.mb_md, a.font_heavy]}>
+          <Trans>Bluesky+</Trans>
+        </Text>
+        <Group
+          label={_(msg`Bluesky+ icons`)}
+          value={currentAppIcon}
+          onChange={onSetAppIcon}>
+          {sets.core.map((icon, i) => (
+            <Row key={icon.id} icon={icon} isEnd={i === sets.core.length - 1}>
+              <AppIcon icon={icon} key={icon.id} size={50} />
+              <RowText>{icon.name}</RowText>
+            </Row>
           ))}
-        </View>
+        </Group>
       </Layout.Content>
     </Layout.Screen>
   )
 }
 
-function useAppIconSets() {
-  const {_} = useLingui()
+function setAppIcon(icon: string) {
+  if (icon === 'default_light') {
+    return getAppIconName(DynamicAppIcon.setAppIcon(null))
+  } else {
+    return getAppIconName(DynamicAppIcon.setAppIcon(icon))
+  }
+}
+
+function getAppIconName(icon: string | false) {
+  if (!icon || icon === 'DEFAULT') {
+    return 'default_light'
+  } else {
+    return icon
+  }
+}
+
+function Group({
+  children,
+  label,
+  value,
+  onChange,
+}: {
+  children: React.ReactNode
+  label: string
+  value: string
+  onChange: (value: string) => void
+}) {
+  return (
+    <Toggle.Group
+      type="radio"
+      label={label}
+      values={[value]}
+      maxSelections={1}
+      onChange={vals => {
+        if (vals[0]) onChange(vals[0])
+      }}>
+      <View style={[a.flex_1, a.rounded_md, a.overflow_hidden]}>
+        {children}
+      </View>
+    </Toggle.Group>
+  )
+}
 
-  return React.useMemo(() => {
-    const defaults = [
-      {
-        id: 'default_light',
-        name: _('Light'),
-        iosImage: () => {
-          return require(`../../../assets/app-icons/ios_icon_default_light.png`)
-        },
-        androidImage: () => {
-          return require(`../../../assets/app-icons/android_icon_default_light.png`)
-        },
-      },
-      {
-        id: 'default_dark',
-        name: _('Dark'),
-        iosImage: () => {
-          return require(`../../../assets/app-icons/ios_icon_default_dark.png`)
-        },
-        androidImage: () => {
-          return require(`../../../assets/app-icons/android_icon_default_dark.png`)
-        },
-      },
-    ]
+function Row({
```

Comment:
Nit: this and the Group feel kinda redundant to split out, could just inline, but nbd

Reply:
Yeah my feeling was this'll make it much easier to add additional sections